File: api/src/main/java/com/leandroadal/vortasks/entities/backup/userprogress/Goals.java
package com.leandroadal.vortasks.entities.backup.userprogress;

import com.leandroadal.vortasks.entities.backup.Backup;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Goals {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Setter(AccessLevel.NONE)
    private String id;

    private Integer dailyProductivity;
    private Integer dailyProductivityProgress;

    private Integer dailyWellBeing;
    private Integer dailyWellBeingProgress;

    private Integer weeklyProductivity;
    private Integer weeklyProductivityProgress;
    
    private Integer weeklyWellBeing;
    private Integer weeklyWellBeingProgress;   
   
    

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "backup_id")
    private Backup userBackup;
    
}

File: app/vortasks/lib/models/goals/goals.dart
class Goals {
  final String id;
  int dailyProductivity;
  int dailyWellBeing;
  int weeklyProductivity;
  int weeklyWellBeing;

  int dailyProductivityProgress;
  int dailyWellBeingProgress;
  int weeklyProductivityProgress;
  int weeklyWellBeingProgress;

  Goals({
    required this.id,
    required this.dailyProductivity,
    required this.dailyWellBeing,
    required this.weeklyProductivity,
    required this.weeklyWellBeing,
    required this.dailyProductivityProgress,
    required this.dailyWellBeingProgress,
    required this.weeklyProductivityProgress,
    required this.weeklyWellBeingProgress,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'dailyProductivity': dailyProductivity,
      'dailyWellBeing': dailyWellBeing,
      'weeklyProductivity': weeklyProductivity,
      'weeklyWellBeing': weeklyWellBeing,
      'dailyProductivityProgress': dailyProductivityProgress,
      'dailyWellBeingProgress': dailyWellBeingProgress,
      'weeklyProductivityProgress': weeklyProductivityProgress,
      'weeklyWellBeingProgress': weeklyWellBeingProgress,
    };
  }

  factory Goals.fromJson(Map<String, dynamic> json) {
    return Goals(
      id: json['id'],
      dailyProductivity: json['dailyProductivity'],
      dailyWellBeing: json['dailyWellBeing'],
      weeklyProductivity: json['weeklyProductivity'],
      weeklyWellBeing: json['weeklyWellBeing'],
      dailyProductivityProgress: json['dailyProductivityProgress'],
      dailyWellBeingProgress: json['dailyWellBeingProgress'],
      weeklyProductivityProgress: json['weeklyProductivityProgress'],
      weeklyWellBeingProgress: json['weeklyWellBeingProgress'],
    );
  }

  Goals copyWith({
    String? id,
    int? dailyProductivity,
    int? dailyWellBeing,
    int? weeklyProductivity,
    int? weeklyWellBeing,
    int? dailyProductivityProgress,
    int? dailyWellBeingProgress,
    int? weeklyProductivityProgress,
    int? weeklyWellBeingProgress,
  }) {
    return Goals(
      id: id ?? this.id,
      dailyProductivity: dailyProductivity ?? this.dailyProductivity,
      dailyWellBeing: dailyWellBeing ?? this.dailyWellBeing,
      weeklyProductivity: weeklyProductivity ?? this.weeklyProductivity,
      weeklyWellBeing: weeklyWellBeing ?? this.weeklyWellBeing,
      dailyProductivityProgress:
          dailyProductivityProgress ?? this.dailyProductivityProgress,
      dailyWellBeingProgress:
          dailyWellBeingProgress ?? this.dailyWellBeingProgress,
      weeklyProductivityProgress:
          weeklyProductivityProgress ?? this.weeklyProductivityProgress,
      weeklyWellBeingProgress:
          weeklyWellBeingProgress ?? this.weeklyWellBeingProgress,
    );
  }
}
